src/OrleansRuntime/Hosting/SiloWrapper.cs

```diff
@@ -0,0 +1,50 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Orleans.Runtime.Hosting
+{
+    internal class SiloWrapper : ISilo
+    {
+        private readonly Silo silo;
+
+        public SiloWrapper(Silo silo, IServiceProvider services)
+        {
+            this.Services = services;
+            this.silo = silo;
+            this.Stopped = silo.SiloTerminated;
+        }
+
+        /// <inheritdoc />
+        public IServiceProvider Services { get; }
+
+        /// <inheritdoc />
+        public Task Stopped { get; }
+
+        /// <inheritdoc />
+        public Task StartAsync(CancellationToken cancellationToken)
```

Comment:
I know that in several places we don't respect that, but since the Start api might be used in a very different place as the rest of Orleans, I would make sure to never throw an exception synchronously. Instead return a faulted task (sure, synchronously too). For that do a try/catch or mark the method async (although you used to get a warning with the latter, not sure if that's the case with the new c#).